src/compute/src/compute_state.rs

```diff
@@ -1399,8 +1400,24 @@ impl IndexPeek {
         let max_result_size = usize::cast_from(max_result_size);
         let count_byte_size = std::mem::size_of::<NonZeroUsize>();
 
-        // Cursor and bound lifetime for `Row` data in the backing trace.
-        let (mut cursor, storage) = oks_handle.cursor();
+        // We have to sort the literal constraints because cursor.seek_key can
+        // seek only forward.
+        peek.literal_constraints
+            .iter_mut()
+            .for_each(|vec| vec.sort());
+        let has_literal_constraints = peek.literal_constraints.is_some();
+        let literals = Option::take(&mut peek.literal_constraints)
+            .into_iter()
+            .flatten();
```

Comment:
What do you think of moving the sorting into `PeekResultIterator::new`? It's an invariant the `PeekResultIterator` requires but doesn't document and cannot check.

Doing this would require changing the type of the `literals` argument to something containing `Vec`, not sure if that causes problems.

Reply:
will do! I'll change it to take a `Option<Vec<>>` and tighten up that unspoken invariant. 👌